mmtk/build.rs

```diff
@@ -13,17 +14,28 @@ fn main() {
     };
 
     // running `make julia_version.h` in $JULIA_PATH/src to generate julia_version.h
-    std::process::Command::new("make")
-        .current_dir(format!("{}/src", julia_dir))
-        .args(["julia_version.h"])
-        .output()
-        .expect("failed to execute process");
+    if !Path::new(format!("{}/src/julia_version.h", julia_dir).as_str()).exists() {
+        std::process::Command::new("make")
+            .current_dir(format!("{}/src", julia_dir))
+            .args(["julia_version.h"])
+            .output()
+            .expect("failed to execute process");
+    }
 
     // runing `make` in $JULIA_PATH/deps to generate $JULIA_PATH/usr/include, in particular libunwind.h
-    std::process::Command::new("make")
-        .current_dir(format!("{}/deps", julia_dir))
-        .output()
-        .expect("failed to execute process");
+    // skip this process if that path already exists since
+    // the .h files could have already beeen generated when building via Makefile
+    if !Path::new(format!("{}/usr/include", julia_dir).as_str()).exists() {
```

Comment:
This could cause problems. The root dir of the source tree could be different from the root dir of the build. For example, if we use `make O=/path-to-my-build/my-julia-build`, the file is generated in `/path-to-my-build/my-julia-build/usr/include` rather than the Julia source dir. We probably need another env var for `BUILDROOT`.

Reply:
The same issue for another line below `.clang_arg(format!("{}/usr/include", julia_dir))`.

Reply:
When using `make O=/path-to-my-build/my-julia-build`, i.e. we build the binding from Julia's Makefile, `JULIA_PATH` is already set to `BUILDROOT` (see [here](https://github.com/udesou/julia/blob/d811d8a4a49b49c195aa28c5e46caa884a28aabf/deps/mmtk_julia.mk#L13) and [here](https://github.com/udesou/julia/blob/d811d8a4a49b49c195aa28c5e46caa884a28aabf/deps/mmtk_julia.mk#L38)).

Reply:
This build script uses both `format!("{}/src/julia.h", julia_dir)` and `format!("{}/usr/include", julia_dir)`. The former one should be the Julia source dir, and the later one should be the build root. I don't know it can work.

Reply:
Should be fixed now.